src/utils/auth.ts

```diff
@@ -0,0 +1,99 @@
+import { address } from "./constants";
+
+class Auth {
+  _address: string;
+  _headers: { [ket: string]: string | number };
+
+  constructor(address: string) {
+    this._address = address;
+    this._headers = {
+      "Content-type": "application/json",
+    };
+  }
+
+  _handleResponse(response: Response) {
+    return response.ok ? response.json() : Promise.reject(response.status);
+  }
+
+  async login(form: {}) {
+    return await fetch(`${this._address}/auth/login`, {
+      method: "POST",
+      headers: this._headers,
+      credentials: "same-origin",
+      body: JSON.stringify(form),
+    } as {}).then(this._handleResponse);
+  }
+
+  async register(form: {}) {
+    return await fetch(`${this._address}/auth/register`, {
+      method: "POST",
+      headers: this._headers,
+      credentials: "same-origin",
+      body: JSON.stringify(form),
+    } as {}).then(this._handleResponse);
+  }
+
+  async logout(token: string) {
+    return await fetch(`${this._address}/auth/logout`, {
+      method: "POST",
+      headers: this._headers,
+      credentials: "same-origin",
+      body: JSON.stringify({ token }),
+    } as {}).then(this._handleResponse);
+  }
+
+  async forgotPassword(email: string) {
+    return await fetch(`${this._address}/password-reset`, {
+      method: "POST",
+      headers: this._headers,
+      credentials: "same-origin",
+      body: JSON.stringify({ email }),
+    } as {}).then(this._handleResponse);
+  }
+
+  async resetPassword(password: string, token: string) {
+    return await fetch(`${this._address}/password-reset/reset`, {
```

Comment:
Надо исправить:
кажется, тут неверный эндпоинт для сброса пароля
все будет работать если убрать
```suggestion
    return await fetch(`${this._address}/password-reset`, {
```